src/QueryScorer.js

```diff
@@ -74,49 +102,136 @@ class QueryScorer {
    *   ordered by score from low to high.  Scores represent edit distance, so
    *   lower scores are better.
    */
-  score(searchString) {
-    // For each word in the query string:
-    //
-    // 1. Get its edit distance from all words in all documents.  While we're
-    //    doing that, keep track of the word's minimum distance per document.
-    // 2. For each document, add the minimum distance computed in the previous
-    //    step to a running sum.  This sum is the document's distance score for
-    //    the query string.
-
-    let searchWords = searchString
+  score(queryString) {
+    let queryWords = queryString
       .trim()
       .split(/\s+/)
       .map(word => word.toLocaleLowerCase());
-    let sumByDoc = new Map();
-    for (let searchWord of searchWords) {
-      let minDistanceByDoc = new Map();
-      for (let [docWord, docs] of this._documentsByWord) {
-        let distance = this._levenshtein(searchWord, docWord);
-        if (distance > this.distanceThreshold) {
-          distance = Infinity;
-        }
-        for (let doc of docs) {
-          minDistanceByDoc.set(
-            doc,
-            Math.min(
-              distance,
-              minDistanceByDoc.has(doc) ? minDistanceByDoc.get(doc) : Infinity
-            )
-          );
-        }
-      }
-      for (let [doc, min] of minDistanceByDoc) {
-        sumByDoc.set(doc, min + (sumByDoc.get(doc) || 0));
-      }
-    }
+    let minDistanceByDoc = this._traverse({ queryWords });
     let results = [];
-    for (let [doc, sum] of sumByDoc) {
-      results.push({ document: doc, score: sum });
+    for (let doc of this._documents) {
+      let distance = minDistanceByDoc.get(doc);
+      results.push({
+        document: doc,
+        score: distance === undefined ? Infinity : distance,
```

Comment:
I'd like for every document to have a score and be present in the results. It's a choice for sure, and we could discard them like you say. In earlier WIP versions of the add-on, I did discard them, and it could sometimes be the case that the results array was empty. (Most recently this happened when I added stop words in the other PR, and when the query consisted only of stop words.) That was surprising in background.js because it couldn't just do `results[0]` to get the top result. It had to null-check it too.